ekwipunek.py

```diff
@@ -0,0 +1,823 @@
+
+    @property
+    def naostrzony(self):
+        return self._naostrzony
+
+
+class BronDwureczna(Bron):
+    """
+    Klasa Broń Dwuręczna - dziedziczy ona po klasie broń, dodaje nowe pole -> naostrzony.
+
+    :param nazwa: nazwa przedmiotu
+    :param wartosc: wartość przedmiotu wyrażana w sztukach złota
+    :param wymagania: wymagane atrybuty do użycia danego przedmiotu
+    :param obrazenia: wartość zadawanych przez przedmiot obrażeń
+    :param naostrzony: bool informujący czy przedmiot jest naostrzony
+    :param efekt: efekt to zamiennik pola uzycie, jak rowniez bonusy ->
+        jego obecnosc informuje iz przedmiot mozna uzyc, a sam efekt informuje
+        o tym co przedmiot robi
+    """
+    def __init__(
+        self,
+        nazwa: str,
+        wartosc: int,
+        wymagania: dict[str, int],
+        obrazenia: float,
+        naostrzony: bool,
+        efekt: dict[str, int] | None = None,
+    ):
+        if efekt is None:
+            efekt = {"obrażenia": obrazenia}
+        super().__init__(nazwa, wartosc, wymagania, obrazenia, efekt)
+        self._naostrzony = naostrzony
+
+    @property
+    def naostrzony(self):
+        return self._naostrzony
+
+
+class Pancerz(Przedmiot):
+    """
+    Klasa pancerzy.
+
+    :param nazwa: nazwa przedmiotu
+    :param wartosc: nazwa przedmiotu wyrażana w sztukach złota
+    :param bron: ochrona przed bronią
+    :param strzaly: ochrona przed strzałami
+    :param ogien: ochrona przed ogniem
+    :param magia: ochrona przed strzałami
+    :param efekt: efekt to zamiennik pola uzycie, jak rowniez bonusy ->
+        jego obecnosc informuje iz przedmiot mozna uzyc, a sam efekt informuje
+        o tym co przedmiot robi
+    """
+
+    def __init__(
+        self,
+        nazwa: str,
+        wartosc: int,
+        bron: int,
+        strzaly: int,
+        ogien: int,
+        magia: int,
+        efekt: dict[str, int] | None = None,
+    ):
+        if efekt is None:
+            efekt = {"Ochrona"}
+        super().__init__(nazwa, wartosc, efekt)
+        self._bron = bron
+        self._strzaly = strzaly
+        self._ogien = ogien
+        self._magia = magia
+
+    @property
+    def bron(self):
+        return self._bron
+
+    @property
+    def strzaly(self):
+        return self._strzaly
+
+    @property
+    def ogien(self):
+        return self._ogien
+
+    @property
+    def magia(self):
+        return self._magia
+
+
+class Magia(Przedmiot):
+    """
+    Klasa Magii.
+
+    :param nazwa: nazwa przedmiotu
+    :param wartosc: nazwa przedmiotu wyrażana w sztukach złota
+    :param mana: koszt many
+    :param efekt: efekt to zamiennik pola uzycie, jak rowniez bonusy ->
+        jego obecnosc informuje iz przedmiot mozna uzyc, a sam efekt informuje
+        o tym co przedmiot robi
+    :param krag: wymagany krąg magii, znajduje się on wtedy gdy mamy do
+        czynienia z runą i jest to jedyny sposób na ich rozróżnienie
+    """
+
+    def __init__(
```

Comment:
Dodaj pole wymagania tutaj, niech pasuje do innych pól. Wymagania powinny opierać się o ilość many. Update powinien byc robiony analogicznie do pola efekt

Reply:
Zastanów się czy pole wymagania nie można by odpowiednio zunifikować gdzieś pod spodem

Reply:
Nie rozumiem zdania - Update powinien być robiony analogicznie do pola efekt

Reply:
### PODAJE PARAMETR MANY KTORY LADUJE W POLU WYMAGANIA

A WYMAGANIA TO BEDZIE ZAWSZE MANA (CZYLI ICH DEFAULTOWO NIE PODAJE)